packages/sdk/docs/classes/CrossChainMessenger.md

```diff
@@ -0,0 +1,1073 @@
+- [getStateBatchAppendedEventByTransactionIndex](CrossChainMessenger.md#getstatebatchappendedeventbytransactionindex)
+- [getStateRootBatchByTransactionIndex](CrossChainMessenger.md#getstaterootbatchbytransactionindex)
+- [getWithdrawalsByAddress](CrossChainMessenger.md#getwithdrawalsbyaddress)
+- [resendMessage](CrossChainMessenger.md#resendmessage)
+- [sendMessage](CrossChainMessenger.md#sendmessage)
+- [toCrossChainMessage](CrossChainMessenger.md#tocrosschainmessage)
+- [waitForMessageReceipt](CrossChainMessenger.md#waitformessagereceipt)
+- [withdrawERC20](CrossChainMessenger.md#withdrawerc20)
+- [withdrawETH](CrossChainMessenger.md#withdraweth)
+
+## Constructors
+
+### constructor
+
+• **new CrossChainMessenger**(`opts`)
+
+Creates a new CrossChainProvider instance.
+
+#### Parameters
+
+| Name | Type | Description |
+| :------ | :------ | :------ |
+| `opts` | `Object` | Options for the provider. |
+| `opts.bridges?` | [`BridgeAdapterData`](../interfaces/BridgeAdapterData.md) | Optional bridge address list. |
+| `opts.contracts?` | [`DeepPartial`](../modules.md#deeppartial)<[`OEContractsLike`](../interfaces/OEContractsLike.md)\> | Optional contract address overrides. |
+| `opts.depositConfirmationBlocks?` | [`NumberLike`](../modules.md#numberlike) | Optional number of blocks before a deposit is confirmed. |
+| `opts.l1BlockTimeSeconds?` | [`NumberLike`](../modules.md#numberlike) | Optional estimated block time in seconds for the L1 chain. |
+| `opts.l1ChainId` | [`NumberLike`](../modules.md#numberlike) | Chain ID for the L1 chain. |
+| `opts.l1SignerOrProvider` | `any` | Signer or Provider for the L1 chain, or a JSON-RPC url. |
+| `opts.l2SignerOrProvider` | `any` | Signer or Provider for the L2 chain, or a JSON-RPC url. |
+
+#### Defined in
+
+[packages/sdk/src/cross-chain-messenger.ts:74](https://github.com/ethereum-optimism/optimism/blob/4e370ce3/packages/sdk/src/cross-chain-messenger.ts#L74)
+
+## Properties
+
+### bridges
+
+• **bridges**: [`BridgeAdapters`](../interfaces/BridgeAdapters.md)
+
+List of custom bridges for the given network.
+
+#### Implementation of
+
+[ICrossChainMessenger](../interfaces/ICrossChainMessenger.md).[bridges](../interfaces/ICrossChainMessenger.md#bridges)
+
+#### Defined in
+
+[packages/sdk/src/cross-chain-messenger.ts:58](https://github.com/ethereum-optimism/optimism/blob/4e370ce3/packages/sdk/src/cross-chain-messenger.ts#L58)
+
+___
+
+### contracts
+
+• **contracts**: [`OEContracts`](../interfaces/OEContracts.md)
+
+Contract objects attached to their respective providers and addresses.
+
+#### Implementation of
+
+[ICrossChainMessenger](../interfaces/ICrossChainMessenger.md).[contracts](../interfaces/ICrossChainMessenger.md#contracts)
+
+#### Defined in
+
+[packages/sdk/src/cross-chain-messenger.ts:57](https://github.com/ethereum-optimism/optimism/blob/4e370ce3/packages/sdk/src/cross-chain-messenger.ts#L57)
```

Comment:
These links to blobs are missing, perhaps they should be relative to `develop`?

Reply:
Hmm will look into this

Reply:
figured out how to make it relative to develop, resolving this